File: backend/src/main/java/ru/bowling/bowlingapp/DTO/MaintenanceScheduleDTO.java
package ru.bowling.bowlingapp.DTO;

import lombok.Builder;
import lombok.Value;

import java.time.LocalDate;

@Value
@Builder
public class MaintenanceScheduleDTO {
    Long scheduleId;
    String maintenanceType;
    LocalDate scheduledDate;
    LocalDate lastPerformed;
    Boolean critical;
}

File: frontend/lib/models/maintenance_schedule_dto.dart
class MaintenanceScheduleDto {
  final int? scheduleId;
  final String? maintenanceType;
  final DateTime? scheduledDate;
  final DateTime? lastPerformed;
  final bool? critical;

  MaintenanceScheduleDto({
    this.scheduleId,
    this.maintenanceType,
    this.scheduledDate,
    this.lastPerformed,
    this.critical,
  });

  factory MaintenanceScheduleDto.fromJson(Map<String, dynamic> json) {
    DateTime? _parseDate(dynamic value) =>
        (value is String && value.isNotEmpty) ? DateTime.tryParse(value) : null;

    return MaintenanceScheduleDto(
      scheduleId: (json['scheduleId'] as num?)?.toInt(),
      maintenanceType: json['maintenanceType'] as String?,
      scheduledDate: _parseDate(json['scheduledDate']),
      lastPerformed: _parseDate(json['lastPerformed']),
      critical: json['critical'] as bool?,
    );
  }
}
